### ISSUE TITLE
Cannot update identity column 'id'.

### ISSUE THREAD

**Issue by eduardofleck** [intel/sails-mssql#29]
Hello,

Thanks for the project. 

I recently moved my sails backend from mysql to mssql and got this problem using /PATCH in my models.

I created a [small project ](https://github.com/eduardofleck/sails-mssql-identity-update) to reproduce the issue.

If I do this

```
     curl --location --request PATCH 'http://localhost:1337/user/1' \
    --header 'Content-Type: application/json' \
    --header 'Cookie: Cookie_2=value; sails.sid=s%3AmFBvgDr9PHBnXgA5T9y4DDciH9QO1Hkb.p%2BrQEPNqGDnGJN9H8%2Fv5ZH1y97ge7NA%2F%2B981P%2BBMFAE' \
    --data-raw '{
    "id": 1,
    "name": "Cool User Update",
    "email": "notme@gmail.com",
    "changePasswordNextLogin": false
    }'
```

I get this error:

```
AdapterError: Unexpected error from database adapter: `sendNativeQuery` failed ("queryFailed"). The database returned an
error when attempting to execute the native query.
Additional data:

{ error:
{ RequestError: Cannot update identity column 'id'.
at handleError (C:\fleck\sails-mssql-identity-update\node_modules\mssql\lib\tedious\request.js:374:15)
at Connection.emit (events.js:198:13)
at Connection.EventEmitter.emit (domain.js:448:20)
at Parser.tokenStreamParser.on.token
(C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\connection.js:832:12)
at Parser.emit (events.js:198:13)
at Parser.EventEmitter.emit (domain.js:448:20)
at Parser.parser.on.token
(C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\token-stream-parser.js:37:14)
at Parser.emit (events.js:198:13)
at Parser.EventEmitter.emit (domain.js:448:20)
at addChunk (C:\fleck\sails-mssql-identity-update\node_modules\readable-stream\lib\_stream_readable.js:298:12)
at readableAddChunk (C:\fleck\sails-mssql-identity-update\node_modules\readable-stream\lib\_stream_readable.js:280:11)
at Parser.Readable.push
(C:\fleck\sails-mssql-identity-update\node_modules\readable-stream\lib\_stream_readable.js:241:10)
at Parser.Transform.push
(C:\fleck\sails-mssql-identity-update\node_modules\readable-stream\lib\_stream_transform.js:139:32)
at doneParsing (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:122:14)
at data (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\infoerror-token-parser.js:48:5)
at call.lineNumber (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\infoerror-token-parser.js:21:19)
at awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:221:7)
at Parser.awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:145:7)
at Parser.readUInt32LE (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:218:10)
at parser.readBVarChar.procName
(C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\infoerror-token-parser.js:20:90)
at readBuffer.data (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:379:9)
at awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:371:7)
at Parser.awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:145:7)
at Parser.readBuffer (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:368:10)
at readUInt8.length (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:378:12)
at awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:165:7)
at Parser.awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:145:7)
at Parser.readUInt8 (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:162:10)
at Parser.readBVarChar (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:377:10)
at parser.readBVarChar.serverName
(C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\infoerror-token-parser.js:19:22)
at readBuffer.data (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:379:9)
at awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:371:7)
at Parser.awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:145:7)
at Parser.readBuffer (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:368:10)
code: 'EREQUEST',
originalError:
{ Error: Cannot update identity column 'id'.
at handleError (C:\fleck\sails-mssql-identity-update\node_modules\mssql\lib\tedious\request.js:372:19)
at Connection.emit (events.js:198:13)
at Connection.EventEmitter.emit (domain.js:448:20)
at Parser.tokenStreamParser.on.token
(C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\connection.js:832:12)
at Parser.emit (events.js:198:13)
at Parser.EventEmitter.emit (domain.js:448:20)
at Parser.parser.on.token
(C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\token-stream-parser.js:37:14)
at Parser.emit (events.js:198:13)
at Parser.EventEmitter.emit (domain.js:448:20)
at addChunk (C:\fleck\sails-mssql-identity-update\node_modules\readable-stream\lib\_stream_readable.js:298:12)
at readableAddChunk (C:\fleck\sails-mssql-identity-update\node_modules\readable-stream\lib\_stream_readable.js:280:11)
at Parser.Readable.push
(C:\fleck\sails-mssql-identity-update\node_modules\readable-stream\lib\_stream_readable.js:241:10)
at Parser.Transform.push
(C:\fleck\sails-mssql-identity-update\node_modules\readable-stream\lib\_stream_transform.js:139:32)
at doneParsing (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:122:14)
at data (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\infoerror-token-parser.js:48:5)
at call.lineNumber (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\infoerror-token-parser.js:21:19)
at awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:221:7)
at Parser.awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:145:7)
at Parser.readUInt32LE (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:218:10)
at parser.readBVarChar.procName
(C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\infoerror-token-parser.js:20:90)
at readBuffer.data (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:379:9)
at awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:371:7)
at Parser.awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:145:7)
at Parser.readBuffer (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:368:10)
at readUInt8.length (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:378:12)
at awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:165:7)
at Parser.awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:145:7)
at Parser.readUInt8 (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:162:10)
at Parser.readBVarChar (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:377:10)
at parser.readBVarChar.serverName
(C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\infoerror-token-parser.js:19:22)
at readBuffer.data (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:379:9)
at awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:371:7)
at Parser.awaitData (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:145:7)
at Parser.readBuffer (C:\fleck\sails-mssql-identity-update\node_modules\tedious\lib\token\stream-parser.js:368:10)
info:
ErrorMessageToken {
name: 'ERROR',
event: 'errorMessage',
number: 8102,
state: 1,
class: 16,
message: 'Cannot update identity column \'id\'.',
serverName: 'librelato-server',
procName: '',
lineNumber: 1 } },
name: 'RequestError',
number: 8102,
lineNumber: 1,
state: 1,
class: 16,
serverName: 'librelato-server',
procName: '',
precedingErrors: [] },
meta: undefined }
at _afterCounting
(C:\fleck\sails-mssql-identity-update\node_modules\waterline\lib\waterline\methods\update-one.js:185:17)
at _afterTalkingToAdapter
(C:\fleck\sails-mssql-identity-update\node_modules\waterline\lib\waterline\methods\count.js:227:16)
at Object.count (C:\fleck\sails-mssql-identity-update\node_modules\sails-mssql\lib\adapter.js:418:14)
at process._tickCallback (internal/process/next_tick.js:68:7)
```

Looks like the update command that is being built contains the identity field...

**Comment by eduardofleck** [comment]
I fixed it on https://github.com/eduardofleck/sails-mssql, but my prettier fucked up the code.
I`ll try to create a pull request just with the needed change

**Comment by davepreston** [comment]
yeah. I recently noticed the same thing, but was working around it by overriding with parseBlueprintOptions.
This is mostly a port of the sails-mysql adapter. mysql doesn't seem to mind sending the id in the update, but sql server certainly does...

I haven't taken the time to deep dive. If you can point out the portion that you altered I could also try updating on my end. (gotta love prettier sometimes :))

**Comment by eduardofleck** [comment]
Hey @davepreston, probably you are right...

In lib/adpter.js, line 548 you will find a comment: 

`// delete values[pk]`

It is almost like a TODO coment... I added below this line of code: 

`if (values.id) delete values.id;`

Thanks :)

**Comment by davepreston** [comment]
@eduardofleck Thanks so much for identifying and narrowing down the bug. Simple fix and it's checked in a published to npm version 2.1.5

thanks again for your help!
